README.md

```diff
@@ -41,55 +43,97 @@ You can help support members of the core team by making a pledge to our Patreon
+[CurseForge page](https://www.curseforge.com/minecraft/mc-mods/lithium). Unless you are using the CurseForge
+client, you should prefer the downloads linked on our Modrinth or GitHub release pages above.
 
-The initial setup may take a few minutes. After Gradle has finished building everything, you can find the resulting
-artifacts in `build/libs`.
+The CurseForge client does not natively support Fabric modding, so you will need to install
+[Jumploader](https://www.curseforge.com/minecraft/mc-mods/jumploader) in order to set up your Fabric environment. Due to
+the extra complexity and startup overhead this workaround adds, we generally do not recommend using this method unless
+you have an existing setup with it.
 
 ---
 
+### Building from sources
+
+Support is not provided for setting up build environments or compiling the mod. We ask that
+users who are looking to get their hands dirty with the code have a basic understanding of compiling Java/Gradle
+projects. The basic overview is provided here for those familiar.
+
+#### Requirements
+
+- JRE 8 or newer (for running Gradle)
+- JDK 8 (optional)
+  - If you neither have JDK 8 available on your shell's path or installed through a supported package manager (such as
+[SDKMAN](https://sdkman.io)), Gradle will automatically download a suitable toolchain from the [AdoptOpenJDK project](https://adoptopenjdk.net/)
+and use it to compile the project. For more information on what package managers are supported and how you can
+customize this behavior on a system-wide level, please see [Gradle's Toolchain user guide](https://docs.gradle.org/current/userguide/toolchains.html).
+- Gradle 6.7 or newer (optional)
+  - The [Gradle wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html#sec:using_wrapper) is provided in
+    this repository can be used instead of installing a suitable version of Gradle yourself. However, if you are building
+    many projects, you may prefer to install it yourself through a suitable package manager as to save disk space and to
+    avoid many different Gradle daemons sitting around in memory.
+
+#### Building with Gradle
+
+Lithium uses a typical Gradle project structure and can be built by simply running the default `build` task. After Gradle
+finishes building the project, you can find the build artifacts (typical mod binaries, and their sources) in
+`build/libs`.
+
+**Tip:** If this is a one-off build, and you would prefer the Gradle daemon does not stick around in memory afterwards,
+try adding the [`--no-daemon` flag](https://docs.gradle.org/current/userguide/gradle_daemon.html#sec:disabling_the_daemon)
+to ensure that the daemon is torn down after the build is complete. However, subsequent builds of the project will
+[start more slowly](https://docs.gradle.org/current/userguide/gradle_daemon.html#sec:why_the_daemon) if the Gradle
+daemon is not available to be re-used.
+
+Build artifacts ending in `api` are for developers compiling against Lithium's API.
```

Comment:
Is this part correct and/or detailed enough? This wasn’t in sodium changes, so I am less sure about it.

Reply:
@DragonEggBedrockBreaking, that is correct. The flag mentioned just before that tells the Gradle daemon to stop running after it compiles the mod. That means every time you build after that it will have to start the daemon again which takes more time, rather than just reconnecting to it. Also the build artifacts ending in api are meant to be used in a dev environment so that is correct.

Reply:
> Also the build artifacts ending in api are meant to be used in a dev environment so that is correct.

I was referring to the API part, sorry if I didn't centre it well (the above part is top-pasted from the sodium README). Since the sodium README goes more into depth about the `-dev` builds, I wanted to be sure that I had added enough detail.